charts/posthog/tests/clickhouse-instance.yaml

```diff
@@ -292,6 +292,26 @@ tests:
             someuser/profile: default
             someuser/quota: default
 
+  - it: handles using existing secret as password
+    set:
+      clickhouse.user: someuser
+      clickhouse.password: somepass
+      clickhouse.existingSecret.name: clickhouse-creds
+      clickhouse.existingSecret.key: password
+    assert:
+      - hasDocuments:
+          count: 1
+      - equal:
+          path: spec.configuration.users
+          value:
+            someuser/k8s_secret_password: clickhouse-creds/password
+            someuser/networks/ip:
```

Comment:
Can we remove the other items from the test that are not the password given that this test is specific to that? I know we don't do that elsewhere, but #kaizen

Reply:
I've re-added the items in [0f2921a](https://github.com/PostHog/charts-clickhouse/pull/506/commits/0f2921ab6c8bbc2a3cdc922484c26833baf23b1d) following the helm unittest errors

When I ran it locally I saw this, so I added them back, unless there's a way to just make it look for a single line that would be preferable.
```
 FAIL  ClickHouse instance      tests/clickhouse-instance.yaml
        - handles using existing secret as password

                - asserts[1] `equal` fail
                        Template:       posthog/templates/clickhouse_instance.yaml
                        DocumentIndex:  0
                        Path:   spec.configuration.users
                        Expected to equal:
                                someuser/k8s_secret_password: clickhouse-creds/password
                        Actual:
                                someuser/k8s_secret_password: clickhouse-creds/password
                                someuser/networks/ip:
                                - 10.0.0.0/8
                                - 172.16.0.0/12
                                - 192.168.0.0/16
                                someuser/profile: default
                                someuser/quota: default
                        Diff:
                                --- Expected
                                +++ Actual
                                @@ -1,2 +1,8 @@
                                 someuser/k8s_secret_password: clickhouse-creds/password
                                +someuser/networks/ip:
                                +- 10.0.0.0/8
                                +- 172.16.0.0/12
                                +- 192.168.0.0/16
                                +someuser/profile: default
                                +someuser/quota: default

        - backup handles using existing secret as password

                - asserts[1] `equal` fail
                        Template:       posthog/templates/clickhouse_instance.yaml
                        DocumentIndex:  0
                        Path:   spec.configuration.users
                        Expected to equal:
                                backupuser/k8s_secret_password: backup-creds/password
                        Actual:
                                admin/networks/ip:
                                - 10.0.0.0/8
                                - 172.16.0.0/12
                                - 192.168.0.0/16
                                admin/password: a1f31e03-c88e-4ca6-a2df-ad49183d15d9
                                admin/profile: default
                                admin/quota: default
                        Diff:
                                --- Expected
                                +++ Actual
                                @@ -1,2 +1,8 @@
                                -backupuser/k8s_secret_password: backup-creds/password
                                +admin/networks/ip:
                                +- 10.0.0.0/8
                                +- 172.16.0.0/12
                                +- 192.168.0.0/16
                                +admin/password: a1f31e03-c88e-4ca6-a2df-ad49183d15d9
                                +admin/profile: default
                                +admin/quota: default
```

Reply:
Yes no problem, I'd missed that it wasn't checking a specific key. Let's get in as is and can leave cleaning up the tests to future selves.

Reply:
There was an escape issue also that I fixed in that commit that was making one e2e test fail. But for the k6 ingestion test and the kubetest postgres I am not too sure if that was related

Reply:
Almost certainly unrelated. Flakiness is on the rise I fear, but not because of this PR.